### DISCUSSION TITLE
Writing Your Own External Data Connectors

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#3880] [Help] [answered]
What would be a good way to set up the dev environment for [writing our own external data connectors](https://docs.budibase.com/references-1/contributing-to-budibase/contribute-data-connectors)?

Any easy way to add a new `private` connector to the docker image? I'm basically thinking of `MY_REST` connector with a few minor esoteric tweaks.

Many thx!

**Comment by mjashanks** [comment, marked as answer]
This shows you how to get setup: https://github.com/Budibase/budibase/blob/HEAD/.github/CONTRIBUTING.md

Doing so would mean that you have to pull changes manually when we build them.

Note that we are about to release V2 of our REST connector, so we might cover your issue soon: https://github.com/Budibase/budibase/issues?q=is%3Aissue+is%3Aopen+%22REST+Connector+V2%22

Writing custom data connectors is definitely something on the horizon, though I don't know when.
